### PR TITLE
feat(cloudformation): update to Resource Specification v4.2.0

### PR DESCRIPTION
**Pull request by rix0rrr**
----

Please read the [contribution guidelines](https://github.com/aws/aws-cdk/blob/master/CONTRIBUTING.md) and follow the pull-request checklist.

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by RomainMuller** [comment]
@shivlaks - I reckon we dialed back on this one and are actually no longer using those names.

File: packages/@aws-cdk/cfnspec/build-tools/spec-diff.ts [hunk 1]
```diff
@@ -158,7 +186,7 @@ async function main() {
     }
 
     if (typeof(update) !== 'object') {
-      throw new Error(`Unexpected change for ${namespace}.${prefix} ${JSON.stringify(update)}`);
+      throw new Error(`Unexpected poopie for ${namespace}.${prefix} ${JSON.stringify(update)}`);
```
Comment (CaerusKaru) [inline packages/@aws-cdk/cfnspec/build-tools/spec-diff.ts]: ?

Comment (rix0rrr) [reply packages/@aws-cdk/cfnspec/build-tools/spec-diff.ts]: Oops

**Comment by rix0rrr** [comment]
Ah, because the *published* packages are still `@stable`, we can't push these changes yet.

We'll have to have a release where the CFN resources are unchanged, but switched to `external` before we can merge this.

**Comment by skinny85** [comment]
> Ah, because the published packages are still @stable, we can't push these changes yet.
>
> We'll have to have a release where the CFN resources are unchanged, but switched to external before we can merge this.

`1.1.0` has been released. Re-triggering the build with a rebase (I can't use merge, as it violates my religious beliefs).

**Pull request merged**